packages/flutter/lib/src/material/dropdown.dart

```diff
@@ -1276,6 +1276,9 @@ class _DropdownButtonState<T> extends State<DropdownButton<T>> with WidgetsBindi
       borderRadius: widget.borderRadius,
     );
 
+    if (!_hasPrimaryFocus) {
```

Comment:
It is safe to call `requestFocus` without checking to make sure it has focus first, so you can just remove this conditional.